vortex-duckdb/src/convert/array/mod.rs

```diff
@@ -38,19 +39,27 @@ pub fn to_duckdb(array: ArrayRef, chunk: &mut dyn WritableVector) -> VortexResul
             .vortex_expect("dict id checked")
             .to_duckdb(chunk)
     } else {
-        array.into_arrow_preferred()?.to_duckdb(chunk)
+        to_arrow_preferred(&array)?.to_duckdb(chunk)
     }
 }
 
 impl ToDuckDB for DictArray {
     fn to_duckdb(&self, chunk: &mut dyn WritableVector) -> VortexResult<()> {
-        to_duckdb(self.values().clone(), chunk)?;
-        let indices =
-            try_cast(self.codes(), &DType::Primitive(U32, NonNullable))?.to_primitive()?;
-        let indices = indices.as_slice::<u32>();
-        let sel = SelectionVector::new_copy(indices);
-        chunk.flat_vector().slice(sel);
-        Ok(())
+        // If the values fit into a single vector, we can efficiently delay the take operation.
+        if self.values().len() <= DUCKDB_STANDARD_VECTOR_SIZE {
```

Comment:
Double checking, should this be `self.values().len()` or `self.values().len() * byte_count_of_the_datatype`?

Reply:
Nope, its length